src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -148,11 +148,19 @@ export class ExtHostTerminal {
 			isExtensionOwnedTerminal: true,
 			useShellEnvironment: withNullAsUndefined(internalOptions?.useShellEnvironment),
 			isSplitTerminal: internalOptions?.isSplitTerminal,
-			target: internalOptions?.target
+			location: !options.location ? undefined : this._getLocation(options.location, internalOptions?.parentTerminal)
 		});
 	}
 
-	public async createExtensionTerminal(isSplitTerminal?: boolean, target?: TerminalLocation, iconPath?: TerminalIcon, color?: ThemeColor): Promise<number> {
+	private _getLocation(location: TerminalLocation | vscode.TerminalEditorLocationOptions | vscode.TerminalSplitLocationOptions, parentTerminal?: ExtHostTerminal): TerminalLocation | vscode.TerminalEditorLocationOptions | { parentTerminal: ExtHostTerminal } | undefined {
```

Comment:
This was a little hard to read, not sure why `parentTerminal` is not in `TerminalSplitLocationOptions` to begin with.

Is this because you're mapping from a `vscode.Terminal` to a `ExtHostTerminal`? In that case `_resolveExtHostLocation` similar to the main thread side would be easier to understand:

```ts
private _resolveExtHostLocation(
location: TerminalLocation | vscode.TerminalEditorLocationOptions | vscode.TerminalSplitLocationOptions
): TerminalLocation | vscode.TerminalEditorLocationOptions | { parentTerminal: ExtHostTerminal }
```